File: lingting-polaris-grpc/src/main/java/live/lingting/polaris/grpc/server/RegisterHook.java

package live.lingting.polaris.grpc.server;

import com.tencent.polaris.api.rpc.InstanceRegisterRequest;
import com.tencent.polaris.api.rpc.InstanceRegisterResponse;

/**
 * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
 */
public interface RegisterHook {

	void beforeRegister(InstanceRegisterRequest instance);

	void afterRegister(InstanceRegisterResponse instance);

}

File: lingting-polaris-grpc/src/main/kotlin/live/lingting/polaris/grpc/server/RegisterHook.kt
package live.lingting.polaris.grpc.server

import com.tencent.polaris.api.rpc.InstanceRegisterRequest
import com.tencent.polaris.api.rpc.InstanceRegisterResponse

/**
 * @author [liaochuntao](mailto:liaochuntao@live.com)
 */
interface RegisterHook {
    fun beforeRegister(instance: InstanceRegisterRequest?)

    fun afterRegister(instance: InstanceRegisterResponse?)
}
